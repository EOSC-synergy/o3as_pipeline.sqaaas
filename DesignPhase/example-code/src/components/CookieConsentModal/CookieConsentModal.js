import React, { useState } from "react";

/**
 * displays an error message
 * @param {Object} props 
 * @param {boolen} props.open -> whether the error message modal should be displayed
 * @param {function} props.onClose -> handles closing of the modal
 * @param {function} props.error -> error function
 * @returns {JSX} a jsx file containing a modal with the given error message
 */
function CookieConsentModal(props) {
    
    let i = props.open;
    i = props.onClose;
    i = props.error;
    

    return <>
    </>;
}

export default CookieConsentModal;