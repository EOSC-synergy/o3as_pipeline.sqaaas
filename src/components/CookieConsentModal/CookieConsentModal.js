import React from "react";
import {useTheme} from "@mui/material/styles";
import {Button, Card, FormControl, Grid, Modal, Typography} from "@mui/material";
import { BiCookie } from "react-icons/bi";

/**
 * displays an error message
 * @param {Object} props
 * @param {boolean} props.isOpen -> whether the error message modal should be displayed
 * @param {function} props.onClose -> handles closing of the modal
 * @returns {JSX.Element} a jsx file containing a modal with the given error message
 */
export default function ErrorMessageModal(props) {

    const style = {
        position: 'absolute',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        bgColor: useTheme().palette.grey[200],
        boxShadow: 24,
    };

    return (
        <>
            <Modal
                open={true}//{props.isOpen}
                onClose={props.onClose}
            >
                <Card sx={style}>
                    <Grid
                        container
                        direction="column"
                    >
                        <Grid
                            item
                            xs={12} sm={12} md={12} lg={12} xl={12}
                            key="cookie"
                            sx={{display: "flex", justifyContent: "center", pt: 5, mb: -3}}
                        >
                            <BiCookie size={100} />
                        </Grid>
                        <Grid
                            item
                            xs={12} sm={12} md={6} lg={6} xl={6}
                            key="text"
                            sx={{px: 5, pt: 5, pb: 3}}>
                            <Typography variant="h3" sx={{textAlign: "center"}}><b>We Use Cookies</b></Typography>
                            <br />
                            <Typography>By using this website, you automatically accept that we use cookies.</Typography>
                        </Grid>
                        <Grid
                            item
                            xs={12} sm={12} md={6} lg={6} xl={6}
                            key="acceptButton"
                            sx={{px: 5, pb: 5, display: "flex", justifyContent: "left"}}
                        >
                            <Button onClick={props.onClose}>Understood</Button>
                        </Grid>
                    </Grid>
                </Card>
            </Modal>
        </>
    );
}

ErrorMessageModal.defaultProps = {
    isOpen: true,
}