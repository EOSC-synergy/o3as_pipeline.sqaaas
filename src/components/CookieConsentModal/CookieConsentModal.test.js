import React from 'react';
import CookieConsentModal from './CookieConsentModal';
import { render } from '@testing-library/react';

describe('test CookieConsentModal component rendering', () => {

    it('renders without crashing', () => {
        render(<CookieConsentModal isOpen={true} onClose={jest.fn()} />);
    });

    it('renders as expected', () => {
        const { baseElement } = render(<CookieConsentModal isOpen={true} onClose={jest.fn()} />);
        expect(baseElement).toMatchSnapshot();
    });

});