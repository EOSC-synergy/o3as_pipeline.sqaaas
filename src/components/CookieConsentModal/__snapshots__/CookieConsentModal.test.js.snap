// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CookieConsentModal component rendering renders as expected 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiModal-root css-79ws1d-MuiModal-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1xb2fui-MuiPaper-root-MuiCard-root"
      tabindex="-1"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-ll0v7f-MuiGrid-root"
        >
          <svg
            fill="currentColor"
            height="100"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21.598 11.064a1.006 1.006 0 0 0-.854-.172A2.938 2.938 0 0 1 20 11c-1.654 0-3-1.346-3.003-2.937.005-.034.016-.136.017-.17a.998.998 0 0 0-1.254-1.006A2.963 2.963 0 0 1 15 7c-1.654 0-3-1.346-3-3 0-.217.031-.444.099-.716a1 1 0 0 0-1.067-1.236A9.956 9.956 0 0 0 2 12c0 5.514 4.486 10 10 10s10-4.486 10-10c0-.049-.003-.097-.007-.16a1.004 1.004 0 0 0-.395-.776zM12 20c-4.411 0-8-3.589-8-8a7.962 7.962 0 0 1 6.006-7.75A5.006 5.006 0 0 0 15 9l.101-.001a5.007 5.007 0 0 0 4.837 4C19.444 16.941 16.073 20 12 20z"
            />
            <circle
              cx="12.5"
              cy="11.5"
              r="1.5"
            />
            <circle
              cx="8.5"
              cy="8.5"
              r="1.5"
            />
            <circle
              cx="7.5"
              cy="12.5"
              r="1.5"
            />
            <circle
              cx="15.5"
              cy="15.5"
              r="1.5"
            />
            <circle
              cx="10.5"
              cy="16.5"
              r="1.5"
            />
          </svg>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-f5ifb1-MuiGrid-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-47suzj-MuiTypography-root"
          >
            <b>
              We Use Cookies!
            </b>
          </h3>
          <br />
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            By using this website, you automatically accept that we use cookies.
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-lc1qsq-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-123d5lk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Disagree
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Agree
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
